src/multiparser.js

```diff
@@ -5,44 +5,38 @@ const docUtils = require("./doc-utils");
 const docBuilders = require("./doc-builders");
 const comments = require("./comments");
 const indent = docBuilders.indent;
+const join = docBuilders.join;
 const hardline = docBuilders.hardline;
 const softline = docBuilders.softline;
 const concat = docBuilders.concat;
 
-function printSubtree(subtreeParser, path, print, options) {
-  const next = Object.assign({}, { transformDoc: doc => doc }, subtreeParser);
-  next.options = Object.assign({}, options, next.options, {
-    parentParser: options.parser,
-    originalText: next.text
-  });
-  if (next.options.parser === "json") {
-    next.options.trailingComma = "none";
-  }
-  const ast = require("./parser").parse(next.text, next.options);
-  const astComments = ast.comments;
-  delete ast.comments;
-  comments.attach(astComments, ast, next.text, next.options);
-  const nextDoc = require("./printer").printAstToDoc(ast, next.options);
-  return next.transformDoc(nextDoc, { path, print });
-}
-
-/**
- * @returns {{ text, options?, transformDoc? } | void}
- */
-function getSubtreeParser(path, options) {
+function printSubtree(path, print, options) {
   switch (options.parser) {
     case "parse5":
-      return fromHtmlParser2(path, options);
+      return fromHtmlParser2(path, print, options);
     case "babylon":
     case "flow":
     case "typescript":
-      return fromBabylonFlowOrTypeScript(path, options);
+      return fromBabylonFlowOrTypeScript(path, print, options);
     case "markdown":
-      return fromMarkdown(path, options);
+      return fromMarkdown(path, print, options);
   }
 }
 
-function fromMarkdown(path, options) {
+function parseAndPrint(passedOptions) {
```

Comment:
You'd need to pass the previous and next options here so the `parentParser` thing works.. see `printSubtree` how it builds `next.options` but gets the `parentParser` from `options`